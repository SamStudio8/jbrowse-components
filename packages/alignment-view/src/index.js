import ViewType from '@gmod/jbrowse-core/pluggableElementTypes/ViewType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import {
  ReactComponent as AlignmentViewReactComponent,
  stateModelFactory as alignmentViewStateModelFactory,
} from './AlignmentView'

export default class extends Plugin {
  install(pluginManager) {
    pluginManager.addViewType(
      () =>
        new ViewType({
          name: 'AlignmentView',
          stateModel: alignmentViewStateModelFactory(pluginManager),
          ReactComponent: AlignmentViewReactComponent,
        }),
    )
  }
}
