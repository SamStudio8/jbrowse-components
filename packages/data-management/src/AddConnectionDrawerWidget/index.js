import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export const ReactComponent = import('./components/AddConnectionDrawerWidget')
export { default as stateModel } from './model'
export const configSchema = ConfigurationSchema('AddConnectionDrawerWidget', {})
