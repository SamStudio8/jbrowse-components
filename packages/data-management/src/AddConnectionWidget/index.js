import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export const ReactComponent = import('./components/AddConnectionWidget')
export { default as stateModel } from './model'
export const configSchema = ConfigurationSchema('AddConnectionWidget', {})
