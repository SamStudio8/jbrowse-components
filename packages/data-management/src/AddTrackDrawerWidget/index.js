import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export const ReactComponent = import('./components/AddTrackDrawerWidget')
export { default as stateModelFactory } from './model'
export const configSchema = ConfigurationSchema('AddTrackDrawerWidget', {})
