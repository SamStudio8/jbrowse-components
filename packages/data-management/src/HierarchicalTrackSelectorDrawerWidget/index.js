import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export const ReactComponent = import('./components/HierarchicalTrackSelector')
export { default as stateModelFactory } from './model'
export const configSchema = ConfigurationSchema(
  'HierarchicalTrackSelectorDrawerWidget',
  {},
)
