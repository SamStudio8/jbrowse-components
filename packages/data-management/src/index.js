import { lazy } from 'react'
import ConnectionType from '@gmod/jbrowse-core/pluggableElementTypes/ConnectionType'
import DrawerWidgetType from '@gmod/jbrowse-core/pluggableElementTypes/DrawerWidgetType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import {
  configSchema as ucscConfigSchema,
  modelFactory as ucscModelFactory,
} from './ucsc-trackhub'
import {
  configSchema as jbrowse1ConfigSchema,
  modelFactory as jbrowse1ModelFactory,
} from './jbrowse1'
import {
  reactComponent as AddTrackReactComponent,
  stateModelFactory as AddTrackStateModelFactory,
  configSchema as AddTrackConfigSchema,
} from './AddTrackDrawerWidget'
import {
  reactComponent as AddConnectionReactComponent,
  stateModel as AddConnectionStateModel,
  configSchema as AddConnectionConfigSchema,
} from './AddConnectionDrawerWidget'
import {
  reactComponent as HierarchicalTrackSelectorReactComponent,
  stateModelFactory as HierarchicalTrackSelectorStateModelFactory,
  configSchema as HierarchicalTrackSelectorConfigSchema,
} from './HierarchicalTrackSelectorDrawerWidget'

export default class extends Plugin {
  install(pluginManager) {
    pluginManager.addConnectionType(
      () =>
        new ConnectionType({
          name: 'UCSCTrackHubConnection',
          configSchema: ucscConfigSchema,
          stateModel: ucscModelFactory(pluginManager),
          displayName: 'UCSC Track Hub',
          description: 'A track or assembly hub in the Track Hub format',
          url: '//genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html#Intro',
        }),
    )

    pluginManager.addConnectionType(
      () =>
        new ConnectionType({
          name: 'JBrowse1Connection',
          configSchema: jbrowse1ConfigSchema,
          stateModel: jbrowse1ModelFactory(pluginManager),
          displayName: 'JBrowse 1 Data',
          description: 'A JBrowse 1 data directory',
          url: '//jbrowse.org/',
        }),
    )

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'HierarchicalTrackSelectorDrawerWidget',
        heading: 'Available Tracks',
        configSchema: HierarchicalTrackSelectorConfigSchema,
        stateModel: HierarchicalTrackSelectorStateModelFactory(pluginManager),
        LazyReactComponent: lazy(() => HierarchicalTrackSelectorReactComponent),
      })
    })

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'AddTrackDrawerWidget',
        heading: 'Add a track',
        configSchema: AddTrackConfigSchema,
        stateModel: AddTrackStateModelFactory(pluginManager),
        LazyReactComponent: lazy(() => AddTrackReactComponent),
      })
    })

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'AddConnectionDrawerWidget',
        heading: 'Add a connection',
        configSchema: AddConnectionConfigSchema,
        stateModel: AddConnectionStateModel,
        LazyReactComponent: lazy(() => AddConnectionReactComponent),
      })
    })
  }
}
