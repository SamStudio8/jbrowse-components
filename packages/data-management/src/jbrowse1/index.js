import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export { default as modelFactory } from './model'

export const configSchema = ConfigurationSchema(
  'JBrowse1Connection',
  {
    name: {
      type: 'string',
      defaultValue: 'nameOfJBrowse1Connection',
    },
    dataDirLocation: {
      type: 'fileLocation',
      defaultValue: { uri: '/path/to/data/dir/' },
    },
    assemblyName: {
      type: 'string',
      defaultValue: '',
    },
    useAssemblySequences: {
      type: 'boolean',
      defaultValue: false,
    },
  },
  { explicitlyTyped: true },
)
