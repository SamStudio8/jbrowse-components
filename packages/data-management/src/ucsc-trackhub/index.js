import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export { default as modelFactory } from './model'

export const configSchema = ConfigurationSchema(
  'UCSCTrackHubConnection',
  {
    name: {
      type: 'string',
      defaultValue: 'nameOfUCSCTrackHubConnection',
      description: 'a unique name for this connection',
    },
    hubTxtLocation: {
      type: 'fileLocation',
      defaultValue: { uri: 'http://mysite.com/path/to/hub.txt' },
      description: 'location of the hub file (usually called hub.txt)',
    },
    assemblyNames: {
      type: 'stringArray',
      defaultValue: [],
      description:
        'names of the assemblies (genomes) in this hub to connect. if left empty, all assemblies will be connected',
    },
    useAssemblySequences: {
      type: 'stringArray',
      defaultValue: [],
      description:
        'name of any assemblies to use to override an existing assembly sequence. probably safe to ignore this.',
    },
  },
  { explicitlyTyped: true },
)
