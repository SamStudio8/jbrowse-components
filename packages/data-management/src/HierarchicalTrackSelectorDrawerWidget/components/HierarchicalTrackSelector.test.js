import React from 'react'
import { render } from 'react-testing-library'
import { createTestEnv } from '@gmod/jbrowse-web/src/JBrowse'
import HierarchicalTrackSelector from './HierarchicalTrackSelector'

window.requestIdleCallback = cb => cb()
window.cancelIdleCallback = () => {}

describe('HierarchicalTrackSelector drawer widget', () => {
  it('renders with just the required model elements', async () => {
    const { session } = await createTestEnv()
    const firstView = session.addView('LinearGenomeView')
    firstView.activateTrackSelector()
    const model = session.drawerWidgets.get('hierarchicalTrackSelector')

    const { container } = render(<HierarchicalTrackSelector model={model} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with a couple of uncategorized tracks', async () => {
    const { session } = await createTestEnv({
      assemblies: [
        {
          assemblyName: 'volvox',
          sequence: {
            adapter: {
              type: 'FromConfigAdapter',
              features: [
                {
                  refName: 'ctgA',
                  uniqueId: 'firstId',
                  start: 0,
                  end: 10,
                  seq: 'cattgttgcg',
                },
              ],
            },
          },
          tracks: [
            {
              configId: 'fooC',
              type: 'BasicTrack',
              adapter: { type: 'FromConfigAdapter', features: [] },
            },
            {
              configId: 'barC',
              type: 'BasicTrack',
              adapter: { type: 'FromConfigAdapter', features: [] },
            },
          ],
        },
      ],
    })
    const firstView = session.addLinearGenomeViewOfAssembly('volvox', {})
    firstView.showTrack(session.configuration.assemblies[0].tracks[0])
    firstView.showTrack(session.configuration.assemblies[0].tracks[1])
    firstView.activateTrackSelector()
    const model = session.drawerWidgets.get('hierarchicalTrackSelector')

    const { container } = render(<HierarchicalTrackSelector model={model} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with a couple of categorized tracks', async () => {
    const { session } = await createTestEnv({
      assemblies: [
        {
          assemblyName: 'volvox',
          sequence: {
            adapter: {
              type: 'FromConfigAdapter',
              features: [
                {
                  refName: 'ctgA',
                  uniqueId: 'firstId',
                  start: 0,
                  end: 10,
                  seq: 'cattgttgcg',
                },
              ],
            },
          },
          tracks: [
            {
              configId: 'fooC',
              type: 'BasicTrack',
              adapter: { type: 'FromConfigAdapter', features: [] },
            },
            {
              configId: 'barC',
              type: 'BasicTrack',
              adapter: { type: 'FromConfigAdapter', features: [] },
            },
          ],
        },
      ],
    })
    const firstView = session.addLinearGenomeViewOfAssembly('volvox', {})
    firstView.showTrack(session.configuration.assemblies[0].tracks[0])
    firstView.showTrack(session.configuration.assemblies[0].tracks[1])
    firstView.tracks[0].configuration.category.set(['Foo Category'])
    firstView.tracks[1].configuration.category.set([
      'Foo Category',
      'Bar Category',
    ])
    firstView.activateTrackSelector()
    const model = session.drawerWidgets.get('hierarchicalTrackSelector')

    const { container } = render(<HierarchicalTrackSelector model={model} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
