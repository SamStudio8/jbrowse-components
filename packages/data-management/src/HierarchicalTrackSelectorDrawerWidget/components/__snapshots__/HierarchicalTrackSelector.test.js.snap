// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HierarchicalTrackSelector drawer widget renders with a couple of categorized tracks 1`] = `
<div
  class="HierarchicalTrackSelector-root-1"
  data-testid="hierarchical_track_selector"
>
  <div
    class="MuiFormControl-root MuiTextField-root HierarchicalTrackSelector-searchBox-2 MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
    >
      Filter Tracks
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root"
            >
              clear
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    
  </div>
  <div
    class="MuiFormGroup-root"
  />
  <button
    class="MuiButtonBase-root MuiFab-root HierarchicalTrackSelector-fab-3 MuiFab-secondary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        add
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`HierarchicalTrackSelector drawer widget renders with a couple of uncategorized tracks 1`] = `
<div
  class="HierarchicalTrackSelector-root-1"
  data-testid="hierarchical_track_selector"
>
  <div
    class="MuiFormControl-root MuiTextField-root HierarchicalTrackSelector-searchBox-2 MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
    >
      Filter Tracks
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root"
            >
              clear
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    
  </div>
  <div
    class="MuiFormGroup-root"
  />
  <button
    class="MuiButtonBase-root MuiFab-root HierarchicalTrackSelector-fab-3 MuiFab-secondary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        add
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`HierarchicalTrackSelector drawer widget renders with just the required model elements 1`] = `
<div
  class="HierarchicalTrackSelector-root-1"
  data-testid="hierarchical_track_selector"
>
  <div
    class="MuiFormControl-root MuiTextField-root HierarchicalTrackSelector-searchBox-2 MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
    >
      Filter Tracks
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root"
            >
              clear
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    
  </div>
  <div
    class="MuiFormGroup-root"
  />
  <button
    class="MuiButtonBase-root MuiFab-root HierarchicalTrackSelector-fab-3 MuiFab-secondary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        add
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
